### PR TITLE
journal: clear the mdJournal branchID after CR completes

### PR DESCRIPTION
**Pull request by strib**
Issue: KBFS-1541

### CODE REVIEW

File: libkbfs/md_journal.go [hunk 1]
```diff
@@ -851,7 +851,10 @@ func (j *mdJournal) clear(
 	}
 
 	if head == (ImmutableBareRootMetadata{}) || head.BID() != bid {
-		// Nothing to do.
+		if j.branchID == bid {
```
Comment (akalin-keybase) [inline libkbfs/md_journal.go]: this should probably be an earlier top-level check, like:

```
if bid == NullBranchId {
  return errors...
}

if bid != j.branchID {
  // Nothing to do.
  return nil
}

if head == ... {
  j.branchID = NullBranchID
  return nil
}
```

Comment (akalin-keybase) [inline libkbfs/md_journal.go]: Also, mind adding some unit tests for clear with `bid != j.branchID` and clear with an empty journal but non-zero `j.branchID`?

File: test/journal_test.go [hunk 2]
```diff
@@ -85,7 +85,7 @@ func TestJournalExclWriteConflict(t *testing.T) {
 }
 
 // bob creates a conflicting file while running the journal.
-func TestJournalCRSimple(t *testing.T) {
+func TestJournalCrSimple(t *testing.T) {
```
Comment (akalin-keybase) [inline test/journal_test.go]: Do you know why this test didn't catch this?

Comment (strib) [inline test/journal_test.go]: Because bob doesn't do any puts after his CR finishes.

File: libkbfs/md_journal.go (hunk 1 above)
Comment (strib) [inline libkbfs/md_journal.go]: There was already a test for the former.  Adding one for the latter.

**Comment by strib** [comment]
CI failure is KBFS-1543,.

**Pull request merged**